enhancements/baremetal/debuggability-of-baremetal-ipi-deployment.md

```diff
@@ -0,0 +1,246 @@
+to make identifying root causes easier.
+
+## Motivation
+
+The goal of this enhancement is to improve the day 1 install experience
+and reduce the perception of complexity in baremetal IPI deployments.
+
+### Goals
+
+- Any deployment that ends in an unsuccessful install must provide the
+  user clear and actionable information to diagnose the problem.
+
+### Non-Goals
+
+- Addressing the underlying causes of the failures are not the goal of
+  this enhancement.
+
+## Proposal
+
+Broadly, deployments fail due to problems encountered during these
+installation activities:
+
+  - Pre-bootstrap (image downloading, manifest creation, etc)
+  - Infrastructure automation (Terraform)
+  - Bootstrap
+  - Bare Metal Host Provisioning (Control Plane and Workers)
+  - Operator Deployment (i.e., those rolled out by CVO)
+
+We believe that since 4.5, pre-bootstrap errors are usually detected,
+and useful information is presented to the user about how to rectify the
+problem, so this enhancement request will focus on failures that occur
+from terraform onward.
+
+### Kinds of deployment failure
+
+#### Infrastructure Automation (Terraform)
+
+Baremetal IPI relies on terraform to provision a libvirt bootstrap
+virtual machine, and bare metal control plane hosts. We use
+terraform-provider-libvirt and terraform-provider-ironic to accomplish
+those goals.
+
+terraform-provider-ironic reports failures when it cannot reach the
+Ironic API, or a control plane host fails to provision. In both cases,
+we do not provide useful information to the user about what to do.
+
+#### Bootstrap Failures
+
+The bootstrap runs a couple of baremetal-specific services, including
+Ironic as well as a utility that populates introspection data for
+the control plane.
+
+Bootstrap typically fails for baremetal when we can't download the
+machine-os image into our local HTTP cache.  Less common, but still
+sometimes seen are that services such as dnsmasq, mariadb, ironic-api,
+ironic-conductor, or ironic-inspector fail.
+
+Failures on bootstrap services rarely result in any indication to the
+user that something went wrong other than that there was a timeout.
+
+The installer has a feature for log gathering on bootstrap failure that
+does not work on baremetal. This should be the first priority, but even
+in this case a user still needs to look into an archive containing many
+logs to identify a failure.
+
+Ideally there would be some mechanism to identify and extract useful
+information and display it to the user.
+
+#### Bare Metal Host Provisioning
+
+Whether the control plane or worker nodes, provisioning of bare metal
+hosts can fail in the same ways, although the communication path to
+provide feedback is different in each case. For the control plane,
+information about failure is presented to the user via terraform. For
+workers, it would be through information on the `BareMetalHost`
+resource, and the baremetal-operator logs.
```

Comment:
This should probably also be reflected as a Degraded status of the `machine-api-operator` or the future metal3 operator.

Reply:
I see you do have this captured later under User Stories, but it wasn't mentioned in this section.